src/types.ts

```diff
@@ -33,6 +33,12 @@ type EnhancedRequestInit<T = string> = Omit<RequestInit, 'body' | 'method'> & {
 
 type ServiceRequestInit<T = string> = Omit<EnhancedRequestInit<T>, 'method'>
 
+type BaseOptions = {
+  headers?: HeadersInit | (() => HeadersInit | Promise<HeadersInit>)
+  requestTransformer?: (request: EnhancedRequestInit) => EnhancedRequestInit
+  responseTransformer?: (response: TypedResponse) => TypedResponse
```

Comment:
Perhaps the transformers should return promises? I had this idea while trying to rewrite the `mergeHeaders` call as a transformer:

```ts
function makeFetcher<T extends string>(baseURL: string | URL, baseOptions: BaseOptions<T> = {}) {
  return async(
    path: T,
    requestInit: EnhancedRequestInit<T> = {},
  ) => {
    const { headers } = baseOptions
    const requestTransformer = baseOptions.requestTransformer ?? identity
    const responseTransformer = baseOptions.responseTransformer ?? identity
    const headerTransformer = async (ri: typeof requestInit) => ({
      ...ri,
      headers: mergeHeaders(
        typeof headers === 'function' ? await headers() : headers ?? {},
        requestInit?.headers ?? {},
      ),
    })
    const url = makeGetApiURL(baseURL)(path)
    const response = await enhancedFetch(
      url,
      await headerTransformer(requestTransformer(requestInit)),
    )
    return responseTransformer(response)
  }
}
```

Reply:
Makes sense